src/components/ui/Text/Text.tsx

```diff
@@ -8,16 +8,29 @@ import { customClassSwitcher } from '~/core';
 
 const COMPONENT_NAME = 'Text';
 
+const tagMap : { [key: string]: string } = {
+    div: 'div',
+    span: 'span',
+    p: 'p',
+    label: 'label',
+  };
+
 export type TextProps = {
     children: React.ReactNode;
     customRootClass?: string;
     className?: string;
+    as?: string;
 } & React.ComponentProps<'p'>;
 
-const Text = ({ children, customRootClass = '', className = '', ...props }: TextProps) => {
+const Text = ({ children, customRootClass = '', className = '', as='p' , ...props }: TextProps) => {
     const rootClass = customClassSwitcher(customRootClass, COMPONENT_NAME);
-
-    return <p className={`${rootClass} ${className}`} {...props}>{children}</p>;
+    
+      
+      const Tag = tagMap[as] || 'p';
```

Comment:
Considering my previous comments, you could just use `as` directly, so this one can be removed